cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -222,12 +223,12 @@ setupWrapper verbosity options mpkg cmd flags extraArgs = do
 --
 determineSetupMethod :: SetupScriptOptions -> BuildType -> SetupMethod
 determineSetupMethod options buildType'
-  | forceExternalSetupMethod options = externalSetupMethod
+  | buildType' == Custom             = externalSetupMethod
```

Comment:
Yes the logic is a bit subtle. Could put in a comment here based on my explanation here https://github.com/haskell/cabal/issues/1938#issuecomment-107456149